cmd/create_test.go

```diff
@@ -1012,4 +951,161 @@ var _ = Describe("Create template", func() {
 			Expect(content).NotTo(ContainSubstring(expectedContent))
 		})
 	})
+
+	Context("given openapi add-on", func() {
+		Context("given only Web variant", func() {
+			It("does NOT contains openapi requirement in .eslintrc.json", func() {
+				cookiecutter := tests.Cookiecutter{
+					AppName: "test-gin-templates",
+					Variant: tests.Web,
+				}
+				cookiecutter.CreateProjectFromGinTemplate(currentTemplatePath)
+				content := tests.ReadFile(".eslintrc.json")
+
+				expectedContent := `plugin:yml/recommended`
+
+				Expect(content).NotTo(ContainSubstring(expectedContent))
+			})
+		})
+
+		Context("given only API variant", func() {
+			It("contains openapi requirement in .eslintrc.json", func() {
+				cookiecutter := tests.Cookiecutter{
+					AppName: "test-gin-templates",
+					Variant: tests.API,
+				}
+				cookiecutter.CreateProjectFromGinTemplate(currentTemplatePath)
+				content := tests.ReadFile(".eslintrc.json")
+
+				expectedContent := `plugin:yml/recommended`
+
+				Expect(content).To(ContainSubstring(expectedContent))
+			})
+		})
+
+		Context("given both variant", func() {
+			It("contains openapi requirement in .eslintrc.json", func() {
+				cookiecutter := tests.Cookiecutter{
+					AppName: "test-gin-templates",
+					Variant: tests.Both,
```

Comment:
I am confused by the conditional check for this option:

https://github.com/nimblehq/gin-templates/blob/a2128788ae57da6a34061541816e43e99681d036/tests/cookiecutter.go#L52-L55

It seems to be set to API variant somehow, even though we have a condition to check hat is not an API. It is odd in understanding this test.

_That change can be done in another PR _

Since Gin is mostly used for API, what do you think of making API the default and Web the rare case? So we could have only one flag to enable Web (--web) and maybe another option to disable the default API (--no-api).

Reply:
@olivierobert I [raised this](https://github.com/nimblehq/gin-templates/pull/106#discussion_r1295341823) with @carryall before. We could follow other web templates by having only 2 variants for templates:
- web + api
- api

Reply:
And it is tracked in https://github.com/nimblehq/gin-templates/issues/111 So all is good